### PR TITLE
`Quiz exercises`: Disable preview button for DnD questions in case no background image is uploaded

### LINKED ISSUE #1392: Drag-And-Drop Question Preview loads forever on empty background image
### Describe the bug
The preview of Drag-And-Drop questions in quiz exercises loads forever if no background image is set.

#### To Reproduce

As an instructor
1. Go to 'Course Management' > 'Exercises' > 'Quiz Exercises'
2. Create a new quiz exercise
3. Add a Drag-And-Drop question
4. Open the preview of this question by clicking the 'Preview' button.

#### Expected behavior

The 'Preview' button should be disabled with a tooltip similar to "A background image must be set before a preview can be generated"

#### Screenshots
preview
loading

### Environment

<details><pre>
 - OS: Windows 10 Pro
 - Browser: Google Chrome
 - Version  81.0.4044.129
</pre></details>

### LINKED ISSUE #5156: Quiz Exercise - DnD questions preview view
### Describe the bug

When you create a DnD question and do not import any image as background, the preview button is clickable anyway. And when you press it, the preview view is loading indefinitely. 
The message says "Please wait until the picture is loaded" but no pictures were uploaded. 

image

### To Reproduce

1. Go to Course Management
2. Click on any course
3. Go to Exercise 
4. Go to quiz exercises 
5. Click on create a new quiz exercise 
6. Click in Add a new Drag and Drop question 
7. Without importing any background, click the preview button. 
8. YOu will see that the view will load indefinitely. 

### Expected behavior

If you create a DnD question and do not import any image as background, the preview button **should not be clickable**. 
Otherwise, when you press it, the preview **should not** say "Please wait until the picture is loaded" the message should say something like "No background has been added". 

### Screenshots

Screenshot 1: The load message
image

### What browsers are you seeing the problem on?

Chrome

### Additional context

I tested this in the `develop` branch release `5.8.2` and also in the production system. 

### Relevant log output

_No response_

### LINKED ISSUE #5158: Quiz exercise - Reset a quiz establishes the visible from one year in advance
### Describe the bug

When you reset a quiz, this shows the "visible from" one year after. See the screenshot below 
image

### To Reproduce

1. Go to Course Management
2. Click on any course
3. Go to Exercise
4. Go to quiz exercises
5. Click on create a new quiz exercise
6. Create any question and save 
7. After passing the deadline or the time to solve it,  reset this quiz. 
8. Verify that the date for "visible from" is one year in advance. 

### Expected behavior

The date for "visible from" should be the actual date. 

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by maximiliansoelch**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Fixes #5156. Fixes #1392.

### Description

To fix #5156, the preview button is disabled until a background image is uploaded. Additionally, a tooltip on the disabled button informs the user that they have to upload a background image first in order to show a preview.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a new quiz exercise
2. Add a new Drag and Drop question
3. Without uploading any background image, the preview button should be disabled and a tooltip should be displayed.
4. Upload a background image5. 
5. The preview button should now be enabled
6. Click the preview button and make sure the actual preview is displayed

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|

### Screenshots

Screenshot showing the disabled button and the corresponding tooltip
Bildschirmfoto 2022-05-31 um 16 09 42 (2)

### CODE REVIEW

**Comment by maximiliansoelch** [comment]
@bottbenj 
Ok, I now see why you have chosen a date in the future and I removed the corresponding change from this commit for now.

But I don't think that this is an optimal solution for the future, as it causes confusion (see #5158).
Because when initially creating a quiz exercise, the release date is set to the current time. Therefore I would expect that the release date is also set to the current time in case I reset a quiz.

File: src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.html [hunk 1]
```diff
@@ -330,11 +330,11 @@ <h3 class="mb-0"><span class="badge bg-warning align-text-top">DnD</span></h3>
         <ng-container *ngIf="!reEvaluationInProgress">
             <hr />
             <div class="question-options">
-                <div class="form-group">
-                    <div class="btn btn-outline-secondary" (click)="togglePreview()">
+                <div ngbTooltip="{{ 'artemisApp.dragAndDropQuestion.disabledPreviewTooltip' | artemisTranslate }}" [disableTooltip]="question.backgroundFilePath !== undefined">
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.html:333]:
```suggestion
                <div ngbTooltip="{{ 'artemisApp.dragAndDropQuestion.disabledPreviewTooltip' | artemisTranslate }}" [disableTooltip]="question.backgroundFilePath">
```
Should also work as far as i know :thinking:

Comment (Dominik-Weinzierl) [reply src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.html:333]: Yes you are right - sorry for that: Tested locally and `[disableTooltip]="!!question.backgroundFilePath"` works as expected :+1: in the following line you can replace `[disabled]="question.backgroundFilePath === undefined"` with `[disabled]="!question.backgroundFilePath"`

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good to me and works as expected :+1: thank you for implementing my suggestions

**Review by willergeorg — APPROVED** [review]
Tested on TS2. Worked fine. Code lgtm as well.

**Pull request merged**